File: 251002/Korti/publisher/src/providers/Article/Article.controller.ts
import {
  Body,
  Controller,
  Delete,
  Get,
  HttpCode,
  HttpStatus,
  Param,
  ParseIntPipe,
  Post,
  Put,
  Query,
} from '@nestjs/common';
import { ArticleService } from './Article.service';
import { ArticleResponseTo } from './Dto/ArticleResponseTo';
import { ArticleRequestTo, UpdateArticleTo } from './Dto/ArticleRequestTo';

@Controller('api/v1.0/articles')
export class ArticleController {
  constructor(private readonly articleService: ArticleService) {}

  @Get()
  @HttpCode(HttpStatus.OK)
  async getAll() {
    const articles = await this.articleService.getAllArticles();
    return articles;
  }

  @Post()
  @HttpCode(HttpStatus.CREATED)
  async create(@Body() body: ArticleRequestTo) {
    const article: ArticleResponseTo =
      await this.articleService.createArticle(body);
    return article;
  }

  @Get(':id')
  @HttpCode(HttpStatus.OK)
  async getOne(@Param('id', ParseIntPipe) idArticle: number) {
    const article = await this.articleService.getArticleById(idArticle);
    return article;
  }

  @Delete(':id')
  @HttpCode(HttpStatus.NO_CONTENT)
  async deleteOne(@Param('id', ParseIntPipe) idArticle: number) {
    await this.articleService.deleteArticle(idArticle);
  }

  @Put()
  @HttpCode(HttpStatus.OK)
  async update(@Body() req: UpdateArticleTo) {
    const article = await this.articleService.updateArticle(req);
    return article;
  }

  @Get()
  @HttpCode(HttpStatus.OK)
  async findArticlesByParams(
    @Query('stickerNames') stickerNames?: string[],
    @Query('stickerIds') stickerIds?: number[],
    @Query('editorLogin') editorLogin?: string,
    @Query('title') title?: string,
    @Query('content') content?: string,
  ): Promise<ArticleResponseTo[]> {
    return this.articleService.findByParams({
      stickerNames,
      stickerIds,
      editorLogin,
      title,
      content,
    });
  }
}

File: 251003/Habrus/publisher/src/main/java/by/bsuir/distcomp/controller/ArticleController.java
package by.bsuir.distcomp.controller;

import by.bsuir.distcomp.dto.request.ArticleRequestTo;
import by.bsuir.distcomp.dto.response.ArticleResponseTo;
import by.bsuir.distcomp.service.ArticleService;
import jakarta.validation.Valid;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

import java.util.List;

@RestController
@RequestMapping(value = "/api/v1.0/articles")
public class ArticleController {
    private final ArticleService articleService;

    public ArticleController(ArticleService articleService) {
        this.articleService = articleService;
    }

    @GetMapping
    ResponseEntity<List<ArticleResponseTo>> getAllArticles() {
        return new ResponseEntity<>(articleService.getAllArticles(), HttpStatus.OK);
    }

    @GetMapping(value = "/{id:\\d+}")
    ResponseEntity<ArticleResponseTo> getArticleById(@PathVariable Long id) {
        return new ResponseEntity<>(articleService.getArticleById(id), HttpStatus.OK);
    }

    @PostMapping
    public ResponseEntity<ArticleResponseTo> createArticle(@RequestBody @Valid ArticleRequestTo articleRequestTo) {
        return new ResponseEntity<>(articleService.createArticle(articleRequestTo), HttpStatus.CREATED);
    }

    @PutMapping
    public ResponseEntity<ArticleResponseTo> updateArticle(@RequestBody @Valid ArticleRequestTo articleRequestTo) {
        return new ResponseEntity<>(articleService.updateArticle(articleRequestTo), HttpStatus.OK);
    }

    @DeleteMapping(value = "/{id:\\d+}")
    public ResponseEntity<ArticleResponseTo> deleteArticle(@PathVariable Long id) {
        articleService.deleteArticle(id);
        return new ResponseEntity<>(HttpStatus.NO_CONTENT);
    }
}
